src/js/binaryen.js-post.js

```diff
@@ -206,35 +206,48 @@ Module['AtomicRMWXchg'] = Module['_BinaryenAtomicRMWXchg']();
 
 // 'Module' interface
 Module['Module'] = function(module) {
-  if (!module) module = Module['_BinaryenModuleCreate']();
-  this['ptr'] = module;
+  assert(!module); // guard against incorrect old API usage
+  var module = Module['_BinaryenModuleCreate']();
+
+  wrapModule(module, this);
+};
+
+// Receives a C pointer to a C Module and a JS object, and creates
+// the JS wrappings on the object to access the C data.
+// This is meant for internal use only, and is necessary as we
+// want to access Module from JS that were perhaps not created
+// from JS.
+function wrapModule(module, self) {
```

Comment:
I actually use this functionality with the old constructor to wrap a C-style reference in a JS-style module, because the compiler uses the C-API so it can cross-compile to JS and WASM. Can we either keep the old API or expose this helper?

Reply:
Makes sense, done.

Reply:
Thanks :)